tests/workflows/basic_prompt_deployment/tests/test_workflow.py

```diff
@@ -156,3 +169,11 @@ def generate_prompt_events(*args: Any, **kwargs: Any) -> Iterator[ExecutePromptE
         "results": expected_outputs,
         "text": "It was hot",
     }
+
+    assert node_events[0].name == "node.execution.initiated"
+    assert (
+        cast(WorkflowParentContext, node_events[0].parent).workflow_definition.model_dump()
+        == WorkflowParentContext(
```

Comment:
[nit] instead of model_dump's, I'm still bullish on overriding the `__eq__` method on CodeResourceDefinition